File: app/src/main/java/jp/searchwakayamatoilet/MainActivity.java
package jp.searchwakayamatoilet;

/**
 * Created by masanori on 2016/07/30.
 * main page activity.
 */
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.google.firebase.analytics.FirebaseAnalytics;

public class MainActivity extends AppCompatActivity{
    private MainPresenter presenter;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FirebaseAnalytics analytics = FirebaseAnalytics.getInstance(this);

        String lastQuery = "";

        if(savedInstanceState != null){
            // 最後に入力されていたQueryがあればセット.
            lastQuery = savedInstanceState.getString(getString(R.string.saveinstance_key_last_query));
            if(lastQuery == null){
                lastQuery = "";
            }
        }
        presenter = new MainPresenter(this, lastQuery);
    }
    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        // 権限リクエストの結果を取得する.
        if (requestCode == R.string.request_permission) {
            presenter.onRequestPermissionsResult(grantResults);
        } else {
            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        }
    }
    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        presenter.onActivityResult(requestCode, resultCode);
    }
    @Override
    public void onPause() {
        // if toilet datas are loading from csv, stop loading.
        presenter.onPaused();
        super.onPause();
    }
    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString(getString(R.string.saveinstance_key_last_query), presenter.getLastQuery());
    }
}

File: app/src/main/java/jp/searchwakayamatoilet/MainActivity.kt
/**
 * Created by Masanori on 2015/12/09.
 * this is main page activity.
 */
package jp.searchwakayamatoilet

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import android.util.Log

class MainActivity : AppCompatActivity() {

    lateinit private var presenter: MainPresenter

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 最後に入力されていたQueryがあればセット,
        val lastQuery = savedInstanceState?.getString(getString(R.string.saveinstance_key_last_query))

        presenter = MainPresenter(this, lastQuery)
    }

    override fun onRequestPermissionsResult(intRequestCode: Int, strPermissions: Array<String>, intGrantResults: IntArray) {
        // 権限リクエストの結果を取得する.
        if (intRequestCode == R.string.request_permission) {
            presenter.onRequestPermissionsResult(intGrantResults)
        } else {
            super.onRequestPermissionsResult(intRequestCode, strPermissions, intGrantResults)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            R.string.request_enable_location -> if (resultCode == Activity.RESULT_OK) {
                // get location data.
                presenter.moveCurrentLocation()
            }
        }
    }

    public override fun onPause() {
        // if toilet datas are loading from csv, stop loading.
        presenter.onPaused()
        super.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(getString(R.string.saveinstance_key_last_query), presenter.strLastQuery)
    }
}
